### DISCUSSION TITLE
TransactionManager.setTransactionTimeout(...)

### DISCUSSION THREAD

**Discussion by famod** [quarkusio/quarkus#22299] [Q&A] [answered]
Hi all,

is something like:
```java
txMgr.setTransactionTimeout(3600);
```
supposed to work or not?

I don't get a `SystemException` so expect the tx to be cancelled by the reaper after one hour, not one minute, but the latter is the case.

Thanks!

**Comment by famod** [comment]
This is with `@TestTransction` btw, /cc @stuartwdouglas. But I don't think that's it...?

**Comment by stuartwdouglas** [comment, marked as answer]
That timeout sets the default timeout for the current thread. So you need to call that method, start a new TX,  do your work, then revert the timeout to clear the thread local state.

This is spec required behavior. If https://github.com/quarkusio/quarkus/pull/21115 is ever merged it will provide a much nicer API.
